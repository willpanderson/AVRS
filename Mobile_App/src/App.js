import * as React from 'react';

import HomePage from './HomePage';
import MainMenu from './MainMenu';

class App extends React.Component {
  state = {
    isLoggedIn: true,
  };

  logInOut = () => {
    let loggedIn = !this.state.isLoggedIn;
    this.setState({
      isLoggedIn: loggedIn,
    });
    console.log(this.state.isLoggedIn);
  };

  render() {
    if (this.state.isLoggedIn === true) {
      return <MainMenu logInChange={this.logInOut} />;
    } else {
      return <HomePage logInChange={this.logInOut} />;
    }
  }
}

export default App;
