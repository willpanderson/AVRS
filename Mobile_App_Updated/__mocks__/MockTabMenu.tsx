import React, {ComponentType} from 'react';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import {NavigationContainer} from '@react-navigation/native';

const Tab = createBottomTabNavigator();

interface propType {
  component: ComponentType<any>;
}

const MockTabMenu: React.FC<propType> = ({component}) => {
  return (
    <NavigationContainer>
      <Tab.Navigator>
        <Tab.Screen
          name="screen"
          component={component}
          options={{
            headerShown: true,
            title: 'Snacks',
            headerStyle: {backgroundColor: '#EFF1F7'},
            headerTitleStyle: {
              fontSize: 30,
              fontWeight: 'bold',
            },
            headerTitleAlign: 'left',
          }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
};

export default MockTabMenu;
