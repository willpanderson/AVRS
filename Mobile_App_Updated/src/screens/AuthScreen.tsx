import {View, Text, StyleSheet, Image, TouchableOpacity} from 'react-native';
import * as React from 'react';
import BackButton from '../components/BackButton';
import InputFields from '../components/InputFields';
// import {AuthContext} from './App';
// import type {StackScreenProps} from '@react-navigation/stack';

// type props = StackScreenProps<>

const AuthScreen = () => {
  const [inputType, setInputType] = React.useState('');

  return (
    <View style={styles.container}>
      <Image style={styles.image} source={require('../img/logo.png')} />

      {inputType !== '' && (
        <View style={styles.backButton}>
          <BackButton
            backPress={() => {
              setInputType('');
            }}
            args={''}
          />
        </View>
      )}

      <View style={styles.buttons}>
        {inputType === 'LOG_IN' && (
          //When user clicks login button
          <InputFields type="Login" />
        )}
        {inputType === 'SIGN_UP' && (
          //When user clicks signup button
          <InputFields type="Sign Up" />
        )}
        {inputType === '' && (
          <>
            <TouchableOpacity
              style={styles.button}
              onPress={() => setInputType('SIGN_UP')}
              testID={'signUp'}>
              <Text style={styles.buttonText}>Sign Up?</Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={styles.button}
              onPress={() => setInputType('LOG_IN')}
              testID={'logIn'}>
              <Text style={styles.buttonText}>Login</Text>
            </TouchableOpacity>
          </>
        )}
      </View>
    </View>
  );
};

export default AuthScreen;

const styles = StyleSheet.create({
  defaultStyle: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#377888',
  },
  image: {
    flex: 2,
    width: 250,
    resizeMode: 'contain',
  },
  buttons: {
    flex: 1,
  },
  button: {
    width: 200,
    height: 50,
    backgroundColor: '#05111B',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    margin: 15,
  },
  buttonText: {
    color: '#DBDDDF',
  },
  backButton: {
    //Absolute has to be applied in the view the component is mounted in
    position: 'absolute',
    top: 50,
    left: 50,
  },
});
