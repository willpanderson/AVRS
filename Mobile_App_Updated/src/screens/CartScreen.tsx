import {View, Text, Pressable, StyleSheet} from 'react-native';
import React, {useState} from 'react';

import CartItem from '../components/CartItem';

import {db} from '../../firebase';
import {doc, setDoc} from 'firebase/firestore';
// import {CartContext} from '../components/MainTabMenu';
import {CartContextType} from '../@types';
import {useCart} from '../functions/CartContext';
import {useIsFocused} from '@react-navigation/native';

export default function ShoppingCartPage() {
  const [orderNum, setOrderNum] = useState('');
  const {cart, emptyCart} = useCart() as CartContextType;

  const isFocused = useIsFocused();

  const placeOrder = () => {
    let order = Math.floor(100000 + Math.random() * 900000);
    let snacks = [];

    for (let i = 0; i < cart.length; i++) {
      for (let j = 0; j < cart[i].count; j++) {
        snacks.push(cart[i].name);
      }
    }
    console.log(cart);
    console.log(snacks);

    setDoc(doc(db, 'orders', String(order)), {Snacks: snacks});
    emptyCart();
    setOrderNum(String(order));
  };

  return (
    <View style={styles.container}>
      <View style={styles.itemsContainer}>
        {isFocused ? (
          cart.map((item, index) => <CartItem key={index} item={index} />)
        ) : (
          <></>
        )}
      </View>
      {orderNum ? (
        <>
          <Text style={styles.orderText}>Order Number:</Text>
          <Text style={styles.orderText}>{orderNum}</Text>
        </>
      ) : (
        <></>
      )}

      {cart.length > 0 ? (
        <View style={styles.buttonContainer}>
          <Pressable
            style={({pressed}) => [
              styles.button,
              pressed ? {opacity: 0.8} : {},
            ]}
            onPress={() => placeOrder()}>
            <Text style={styles.buttonText}>Place Order</Text>
          </Pressable>
        </View>
      ) : (
        <View style={styles.container}>
          <Text>Cart is empty</Text>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
  },
  headerContainer: {
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  itemsContainer: {
    flex: 4,
    justifyContent: 'flex-start',
    width: '100%',
  },
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  header: {
    margin: 10,
    fontSize: 30,
    fontWeight: 'bold',
  },
  orderText: {
    fontSize: 20,
    margin: 10,
  },
  button: {
    width: 200,
    height: 50,
    backgroundColor: '#05111B',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    margin: 15,
  },
  buttonText: {
    color: '#DBDDDF',
  },
});
