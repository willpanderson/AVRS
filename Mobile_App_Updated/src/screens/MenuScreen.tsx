import React from 'react';
import {View, StyleSheet} from 'react-native';
import {db} from '../../firebase';
import {collection, getDocs} from 'firebase/firestore';
import MenuItem from '../components/MenuItem';
import {CartContextType, SnackType} from '../@types';
import {useCart} from '../functions/CartContext';

export default function MenuPage() {
  const [snacks, setSnacks] = React.useState<SnackType[]>([]);

  const {cart, addToCart, addCartItem} = useCart() as CartContextType;

  React.useEffect(() => {
    const snacksRef = collection(db, 'snacks');
    const fetchData = async () => {
      let snacksTemp: SnackType[] = [];
      const snapshot = await getDocs(snacksRef);
      snapshot.forEach(doc => {
        console.log(doc.id, ' => ', doc.data());
        snacksTemp.push({
          ...(doc.data() as SnackType),
        });
      });
      setSnacks(snacksTemp);
      console.log('finished');
    };
    fetchData();
  }, []);

  const onSnackPress = (key: string) => {
    console.log('on press', key);
    let temp = {
      name: '',
      count: 0,
    };
    temp.name = key;
    temp.count = 1;

    // Check if the item already is in the list
    // If yes, then add to count
    if (cart.some(item => item.name === key)) {
      for (let i = 0; i < cart.length; i++) {
        if (cart[i].name === key) {
          addCartItem(i);
        }
      }
    } else {
      addToCart(temp);
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.snacksContainer}>
        {snacks.map((snack, index) => (
          <MenuItem snack={snack} onSnackPress={onSnackPress} key={index} />
        ))}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  imageStyle: {
    width: 100,
    height: 100,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    backgroundColor: '#EFF1F7',
  },
  headerContainer: {
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  snacksContainer: {
    flex: 5,
    width: '100%',
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  menuItem: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 150,
    height: 150,
    margin: 10,
    backgroundColor: 'white',
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 10,
  },
  header: {
    margin: 10,
    fontSize: 30,
    fontWeight: 'bold',
  },
  navBar: {
    flexDirection: 'row',
    width: '90%',
    justifyContent: 'space-evenly',
  },
});
