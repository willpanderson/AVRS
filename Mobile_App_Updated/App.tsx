import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import EncryptedStorage from 'react-native-encrypted-storage';
import AuthScreen from './src/screens/AuthScreen';
import MainTabMenu from './src/components/MainTabMenu';
import {AuthContextType, RootStackParamList} from './src/@types';
import {
  createAccount,
  handleSignOut,
  logIn,
} from './src/functions/firebaseHelperFunctions';
import LoadingScreen from './src/screens/LoadingScreen';

export const AuthContext = React.createContext<AuthContextType | null>(null);

const Stack = createStackNavigator<RootStackParamList>();

const App = () => {
  const [state, dispatch] = React.useReducer(
    (prevState: any, action: {type: any; token: any}) => {
      switch (action.type) {
        case 'RESTORE_TOKEN':
          return {
            ...prevState,
            userToken: action.token,
            isLoading: false,
          };
        case 'SIGN_IN':
          return {
            ...prevState,
            isSignout: false,
            userToken: action.token,
          };
        case 'SIGN_OUT':
          return {
            ...prevState,
            isSignout: true,
            userToken: null,
          };
      }
    },
    {
      isLoading: true,
      isSignout: false,
      userToken: null,
    },
  );

  React.useEffect(() => {
    const bootStrapAsync = async () => {
      let userToken;
      try {
        userToken = await EncryptedStorage.getItem('usser_session');
        if (userToken !== undefined) {
          console.log(userToken);
        }
      } catch (error) {
        console.log(error);
      }
      dispatch({type: 'RESTORE_TOKEN', token: userToken});
    };

    bootStrapAsync();
  }, []);

  const storeUserSession = async () => {
    console.log('Setting user session');
    try {
      await EncryptedStorage.setItem(
        'usser_session',
        JSON.stringify({
          token: 'ACCESS_TOKEN',
        }),
      );
      console.log('Store token');
    } catch (error) {
      console.log(error);
    }
  };

  const authContext = React.useMemo(
    () => ({
      signIn: async (email: string, password: string) => {
        console.log(email + password);
        let value = await logIn(email, password);
        if (value) {
          dispatch({type: 'SIGN_IN', token: 'bob'});
          storeUserSession();
        }
        console.log(value);
      },
      signOut: () => {
        try {
          EncryptedStorage.removeItem('usser_session');
          console.log('removes session');
        } catch (error) {
          console.log(error);
        }
        handleSignOut();
        dispatch({
          type: 'SIGN_OUT',
          token: undefined,
        });
      },
      signUp: async (email: string, password: string) => {
        console.log(email + password);
        let value = await createAccount(email, password);
        if (value) {
          dispatch({type: 'SIGN_IN', token: 'bob'});
          storeUserSession();
        }
      },
    }),
    [],
  );

  return (
    <AuthContext.Provider value={authContext}>
      <NavigationContainer>
        <Stack.Navigator>
          {state.isLoading ? (
            <Stack.Screen name="splash" component={LoadingScreen} />
          ) : state.userToken == null ? (
            <Stack.Screen
              name="auth"
              component={AuthScreen}
              options={{headerShown: false}}
            />
          ) : (
            <Stack.Screen
              name="tabMenu"
              component={MainTabMenu}
              options={{headerShown: false}}
            />
          )}
        </Stack.Navigator>
      </NavigationContainer>
    </AuthContext.Provider>
  );
};

export default App;
