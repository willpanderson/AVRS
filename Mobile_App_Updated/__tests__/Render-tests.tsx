/**
 * @format
 */

import 'react-native';
import React from 'react';
import App, {AuthContext} from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import AuthScreen from '../src/screens/AuthScreen';
import CartScreen from '../src/screens/CartScreen';
import MenuScreen from '../src/screens/MenuScreen';
import SettingsScreen from '../src/screens/SettingsScreen';
import CartProvider from '../src/functions/CartContext';
import MockTabMenu from '../__mocks__/MockTabMenu';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('Testing if screens will render', () => {
  it('App renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('AuthScreen renders correctly', () => {
    const tree = renderer.create(<AuthScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('CartScreen renders correctly', () => {
    // const useIsFocused = jest.fn();

    jest.mock('@react-navigation/native', () => ({
      useIsFocused: () => true,
    }));

    // const useIsFocussed = jest.fn();

    jest.mock('@react-navigation/native', () => {
      const actualNav = jest.requireActual('@react-navigation/native');
      return {
        ...actualNav,
        useIsFocussed: jest.fn(),
      };
    });

    const tree = renderer
      .create(
        <CartProvider>
          <MockTabMenu component={CartScreen} />
        </CartProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('MenuScreen renders correctly', () => {
    const tree = renderer
      .create(
        <CartProvider>
          <MenuScreen />
        </CartProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const signIn = jest.fn();
  const signUp = jest.fn();
  const signOut = jest.fn();
  it('SettingsScreen renders correctly', () => {
    const tree = renderer
      .create(
        <AuthContext.Provider value={{signIn, signUp, signOut}}>
          <SettingsScreen />
        </AuthContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
