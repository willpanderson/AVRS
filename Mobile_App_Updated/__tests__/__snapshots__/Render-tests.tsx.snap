// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if screens will render App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  splash
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 5,
                              "justifyContent": "flex-end",
                            }
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#05111B",
                                  "borderRadius": 10,
                                  "height": 50,
                                  "justifyContent": "center",
                                  "margin": 15,
                                  "width": 200,
                                },
                                Object {},
                              ]
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#DBDDDF",
                                }
                              }
                            >
                              Sign Out
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Testing if screens will render AuthScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#377888",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "default": "",
      }
    }
    style={
      Object {
        "flex": 2,
        "resizeMode": "contain",
        "width": 250,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#05111B",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 1,
          "width": 200,
        }
      }
      testID="signUp"
    >
      <Text
        style={
          Object {
            "color": "#DBDDDF",
          }
        }
      >
        Sign Up?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#05111B",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 1,
          "width": 200,
        }
      }
      testID="logIn"
    >
      <Text
        style={
          Object {
            "color": "#DBDDDF",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing if screens will render CartScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text>
                Cart is empty
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "#EFF1F7",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "justifyContent": "center",
                    "marginStart": 0,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 30,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Snacks
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Snacks, tab, 1 of 1"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "rgb(0, 122, 255)",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            >
              ⏷
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#8E8E8F",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            >
              ⏷
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "rgb(0, 122, 255)",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Snacks
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Testing if screens will render MenuScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EFF1F7",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 5,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`Testing if screens will render SettingsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 5,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#05111B",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "margin": 15,
            "width": 200,
          },
          Object {},
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#DBDDDF",
          }
        }
      >
        Sign Out
      </Text>
    </View>
  </View>
</View>
`;
